pkg/dogstatsd/parser.go

```diff
@@ -279,7 +280,14 @@ func parseMetricMessage(message []byte) (*metrics.MetricSample, error) {
 	}
 
 	metricName := string(rawName)
-
+	if config.Datadog.GetString("statsd_metric_namespace") != "" {
+		metricPrefix := config.Datadog.GetString("statsd_metric_namespace")
```

Comment:
this could be done only once in `server.go:Server.worker` and given as a parameter to `parseMetricMessage`. That would avoid checking the suffix over and over. Plus we don't add the namespace if it does not end with a `.`.